hudi-common/src/main/java/org/apache/hudi/common/util/ParquetUtils.java

```diff
@@ -103,6 +120,42 @@
     return rowKeys;
   }
 
+  /**
+   * Read the rows with record key and partition path from the given parquet file
+   *
+   * @param filePath      The parquet file path.
+   * @param configuration configuration to build fs object
+   * @return Set Set of row keys matching candidateRecordKeys
+   */
+  public static List<Pair<Pair<String, String>, Option<HoodieRecordLocation>>> fetchRecordKeyPartitionPathFromParquet(Configuration configuration, Path filePath,
+                                                                                                                      String baseInstantTime,
+                                                                                                                      String fileId) {
+    List<Pair<Pair<String, String>, Option<HoodieRecordLocation>>> rows = new ArrayList<>();
+    try {
+      if (!filePath.getFileSystem(configuration).exists(filePath)) {
+        return new ArrayList<>();
+      }
+      Configuration conf = new Configuration(configuration);
+      conf.addResource(FSUtils.getFs(filePath.toString(), conf).getConf());
+      Schema readSchema = HoodieAvroUtils.getRecordKeyPartitionPathSchema();
+      AvroReadSupport.setAvroReadSchema(conf, readSchema);
+      AvroReadSupport.setRequestedProjection(conf, readSchema);
+      ParquetReader reader = AvroParquetReader.builder(filePath).withConf(conf).build();
```

Comment:
As of now, I don't do the lazy record iterator. Thought will make that in a diff patch as I want to get this out sooner.

Reply:
I think we can fix it in this patch itself.. its a critical aspect